tooling/backend_interface/src/lib.rs

```diff
@@ -40,11 +40,16 @@ pub enum BackendError {
     #[error("Backend binary does not exist")]
     MissingBinary,
 
-    #[error("The backend responded with malformed data: {0:?}")]
-    MalformedResponse(Vec<u8>),
+    #[error("The backend responded with a malformed UTF8 byte vector: {0:?}")]
+    InvalidUTF8Vector(Vec<u8>),
 
-    #[error("The backend encountered an error")]
-    CommandFailed(Vec<u8>),
+    #[error(
+        "The backend responded with a unexpected number of bytes. Expected: {0:} but got {1:?}"
```

Comment:
```suggestion
        "The backend responded with a unexpected number of bytes. Expected: {0} but got {.1.len()} ({1:?})",
```

Not sure if this syntax works but currently we're comparing a length to the actual value of the bytes received. We should print the length of the received bytes and then print the actual value afterwards.

Reply:
The syntax is not valid, though I pushed a version which uses an old syntax which works. tested by doing:


    #[test]
    fn print_err() {
        println!("{}", BackendError::UnexpectedNumberOfBytes(10, vec![0, 1, 2, 3, 4]));
    }